FLEXIN-737: Update logic for filename truncation

mdeshpande12 opened the pull request:
- Updated definition for Notification interface
- Updated logic for shortening for filename for it to truncate the filename only when filename characters are more than 20 characters
- Moved file related logic to single `file.ts` utility file.



**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

deepak-thomas-t commented on cypress.config.ts:
```diff
@@ -24,7 +24,7 @@ export default defineConfig({
         downloadsFolder: "cypress/downloads",
         trashAssetsBeforeRuns: true,
-        chromeWebSecurity: false,
+        chromeWebSecurity: true,
```
is this needed?

mdeshpande12 replied:
this is not needed here. I just cherry picked this commit and added it to this branch, just to make sure that the e2e tests are running fine. This commit will go away once this PR merges -> https://github.com/twilio/twilio-webchat-react-app/pull/133

deepak-thomas-t commented on src/__tests__/notifications.test.ts:
```diff
@@ -0,0 +1,14 @@
+describe("notifications", () => {
+    it("should shorten filename if the filename char is too long", () => {
+        const longFileName = "Profile-20230724T08574.txt";
+        const truncatedName = notifications.shortenFileName(longFileName);
```
can you add a test for a file name with 20 character and 19 character

mdeshpande12 replied:
That's a good catch, the logic should be upto 20 characters. I'll change the comparison from 'less than' to 'less than or equal to 20'. Also, I think test for 19 characters is covered with the short file name test.

AnudeepChPaul commented on src/notifications.ts:
```diff
@@ -9,10 +9,18 @@ import { Notification } from "./store/definitions";
-    return `${filename.substr(0, maxChar)}[...]${fileExtension || ""}`;
+    if (filename.length <= maxChar) return name;
+
+    return `${filename.substring(0, 14)}[...]${filename.substring(
```
Is 14 the recommended by PM or anyone ?

Else I'll suggest to reduce it further. like 7

AnudeepChPaul replied:
Also, take this variable into a top level declaration.

like at top of the file,

`const MAX_DISPLAYED_CHAR = 7`

and at line 20

`return `${filename.substring(0, MAX_DISPLAYED_CHAR)}`

mdeshpande12 replied:
@AnudeepChPaul yes this is recommended & confirmed by Stu and Shilpa as well.

AnudeepChPaul replied:
Then move this 14 to top level declaration. 

```const MAX_DISPLAYED_CHAR = 14 ```

at line 20,
```return ${filename.substring(0, MAX_DISPLAYED_CHAR)}```

mdeshpande12 replied:
pushed a commit for this.

deepak-thomas-t commented on src/notifications.ts:
```diff
@@ -8,11 +8,20 @@ import { Notification } from "./store/definitions";
+    return `${filename.substring(0, MAX_DISPLAYED_CHAR_FOR_FIRST_PART)}[...]${filename.substring(
+        filename.length - 6,
+        filename.length - 1
+    )}${fileExtension}`;
```
wouldn't `filename.substring(-5)` be enough instead of `filename.substring(
        filename.length - 6,
        filename.length - 1
    )`

mdeshpande12 replied:
somehow this file got skipped in the commit. Actually, `filename.substring(filename.length-5)` is the shortform that would work.

deepak-thomas-t commented on src/notifications.ts:
```diff
@@ -8,11 +8,20 @@ import { Notification } from "./store/definitions";
+const MAX_DISPLAYED_CHAR_FOR_FIRST_PART = 14;
+const MAX_DISPLAYED_CHAR_FOR_SECOND_PART = 5;
 
-const shortenFileName = (string: string, maxChar = 50) => {
```
This function is not related to notification right ?? It would be better move it to `src/utils`

mdeshpande12 replied:
pushed a commit for this

mdeshpande12 commented:
Added following changes:

- Moved file related logic to single file.ts utility file.
- Reverted back the logic from commit [a7a2dc127300e8a42f21d967ce55bfaec74212d5](https://github.com/twilio/twilio-webchat-react-app/pull/133/commits/a7a2dc127300e8a42f21d967ce55bfaec74212d5); as I noticed issues in parsing of Map DataStructure in localStorage

deepak-thomas-t reviewed (approved):
LGTM
